tests/jerry/fail/regression-test-issue-3554.js

```diff
@@ -0,0 +1,15 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(){try{}catch(e){function e
```

Comment:
Could we extend the test case to be a syntactically correct code? In the current state it could fail with two reasons:
* Error as the code is not complete (unexpected end of input)
* or with local variable re-declaration error.

If the test code is complete then we can avoid triggering the first error case.

Reply:
I added another test, which covers the syntax and runtime. It should be enough.